lib/hotel.rb

```diff
@@ -0,0 +1,162 @@
+
+require_relative "room"
+require_relative "reservation"
+require_relative "hotelblock"
+
+module HotelGroup
+  class Hotel
+    attr_accessor :id, :rooms, :reservations, :blocks
+
+    def initialize(directory: "./csv")
+      @id = 1
+      @rooms = Room.load_all(directory: directory)
+      @reservations = Reservation.load_all(directory: directory)
+      @blocks = HotelBlock.load_all(directory: directory)
+
+      connect_reservations
+      connect_blocks
+    end
+
+    def connect_blocks
+      blocks.each do |block|
+        room_array = []
+        block.rooms.each do |room_id|
+          room_obj = find_room(room_id)
+          room_array << room_obj
+          room_obj.add_block_id(block.id)
+          room_obj.block_price = room_obj.price - room_obj.price * block.discount
+        end
+        block.rooms = room_array
+      end
+      return blocks
+    end
+
+    def connect_reservations
+      reservations.each do |res|
+        room = find_room(res.room)
+        res.connect(room)
+        room.connect(res)
+      end
```

Comment:
This sort of structure makes sense if you want to look things up from either end, but is it really needed here?